modules/afamqp/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+external_or_find_package(RabbitMQ)
+
+if (RabbitMQ_FOUND)
+  option(ENABLE_AFAMQ "Enable afamqp module" ON)
+else()
+  option(ENABLE_AFAMQ "Enable afamqp module" OFF)
+endif()
+
+if (ENABLE_AFAMQ)
+
+set(AFAMQ_HEADERS
+    "afamqp-parser.h"
+    "afamqp.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/afamqp-grammar.h"
+)
+
+set(AFAMQ_SOURCES
+    "afamqp-parser.c"
+    "afamqp.c"
+    "${CMAKE_CURRENT_BINARY_DIR}/afamqp-grammar.c"
+)
+
+generate_y_from_ym(modules/afamqp/afamqp-grammar)
+
+bison_target(AMQPGrammar
+    ${CMAKE_CURRENT_BINARY_DIR}/afamqp-grammar.y
+    ${CMAKE_CURRENT_BINARY_DIR}/afamqp-grammar.c
+    COMPILE_FLAGS ${BISON_FLAGS})
+
+include_directories (${CMAKE_CURRENT_BINARY_DIR})
+include_directories (${CMAKE_CURRENT_SOURCE_DIR})
```

Comment:
When using external rabbitmq, I miss the header include directory for it. Why can we avoid setting it?

Reply:
We shall not. Fixed.

The reason why it worked in CI:

**RabbitMQ_SOURCE=system**
The compiler (for now clang) default include directories are:
```
#include "..." search starts here:
#include <...> search starts here:
 /usr/bin/../lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/c++/6.3.0
 /usr/bin/../lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/x86_64-linux-gnu/c++/6.3.0
 /usr/bin/../lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/c++/6.3.0/backward
 /usr/include/clang/4.0.0/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
```
The installed headers of `librabbitmq` are in `/usr/include`.

**RabbitMQ_SOURCE=internal**
In this case our cmake installs it into the `${CMAKE_CURRENT_BINARY_DIR}`, which is included also, thus not causing compile error.